src/core/proj/qgscoordinatereferencesystem.cpp

```diff
@@ -833,6 +837,9 @@ bool QgsCoordinateReferenceSystem::hasAxisInverted() const
 
 QList<Qgis::CrsAxisDirection> QgsCoordinateReferenceSystem::axisOrdering() const
 {
+  if ( type() == Qgis::CrsType::Compound )
+    return horizontalCrs().axisOrdering() + verticalCrs().axisOrdering();
```

Comment:
This looks fine to me. @rouault I'm interested in your insights though -- do you think there's any risk of a compound CRS ever needing Z / X / Y or Z / Y / X ordering? :thinking:

Reply:
There might be compound CRS that are horizontal + temporal, or horizontal + vertical + temporal, but those are super esoteric. The proposed code should be fine for now